seastar/src/config_and_start_seastar.hh

```diff
@@ -0,0 +1,30 @@
+#pragma once
+
+#include "rust/cxx.h"
+#include <seastar/core/app-template.hh>
+
+namespace seastar {
+
+using seastar_options = app_template::seastar_options;
+
+std::unique_ptr<seastar_options> new_options();
+
+rust::Str get_name(const std::unique_ptr<seastar_options>& opts);
+
+rust::Str get_description(const std::unique_ptr<seastar_options>& opts);
+
+uint32_t get_smp(const std::unique_ptr<seastar_options>& opts);
+
+void set_name(const std::unique_ptr<seastar_options>& opts, const rust::Str name);
+
+void set_description(const std::unique_ptr<seastar_options>& opts, const rust::Str description);
+
+void set_smp(const std::unique_ptr<seastar_options>& opts, const uint32_t smp);
```

Comment:
Why not pass `seastar_options&`?

Reply:
Rust doesn't have access to that type directly, as it's opaque. It gets it under a `UniquePtr`, so it must transfer it through a `UniquePtr` as well.

Reply:
Indeed, that'd even make for shorter code.